lib/Doctrine/Common/Collections/Collection.php

```
@@ -212,6 +212,19 @@ public function next();
      */
     public function exists(Closure $p) : bool;
 
+    /**
+     * Returns the first element of this collection that satisfies the predicate p.
+     *
+     * @param Closure $p The predicate.
+     *
+     * @return mixed The first element respecting the predicate,
```

What if the first element respecting the predicate is literally `false` itself? Wouldn't it be better to throw an exception that recommends to call whatever method allows to check that there will be a result? (`exists`?)

---

I don't like to require an `exists` call, it add complexity to the code and it lower performance (the search is made twice).

Collection is not made to store boolean.
All the method `current/next/first/last/...` return `T|false`.

But maybe, It's better to return `null`.
The method `get` is returning `T|null`.

---

I changed for `null` if not found.

---

>  All the method current/next/first/last/... return T|false.

Ok, if this is an issue, it's an issue of doctrine/collections, not this PR